src/ui.rs

```diff
@@ -3047,40 +3047,60 @@ impl Component for SectionLineView<'_> {
     }
 
     fn draw(&self, viewport: &mut Viewport<Self::Id>, x: isize, y: isize) {
-        const NEWLINE_ICON: &str = "⏎";
-        let Self { line_key: _, inner } = self;
+        fn replacement_characters(character: char) -> Option<&'static str> {
+            match character {
+                // Characters end up writing over each-other and end up
+                // displaying incorrectly if ignored. Replacing tabs
+                // with a known length string fixes the issue for now.
+                '\t' => Some("→   "),
+                '\n' => Some("⏎"),
+                _ => None,
+            }
+        }
+
+        /// Split the line into a sequence of [`Span`]s where control characters are
+        /// replaced with styled [`Span`]'s and push them to the [`spans`] argument.
+        fn push_spans_from_line<'line>(line: &'line str, spans: &mut Vec<Span<'line>>) {
+            const CONTROL_CHARACTER_STYLE: Style = Style::new().fg(Color::DarkGray);
+
+            let mut last_index = 0;
+            // Find index of the start of each character to replace
+            for (idx, char) in line.match_indices(|char| replacement_characters(char).is_some()) {
+                // Push the string leading up to the character and the styled replacment string
+                if let Some(replacment_string) =
+                    char.chars().next().and_then(replacement_characters)
+                {
+                    spans.push(Span::raw(&line[last_index..idx]));
+                    spans.push(Span::styled(replacment_string, CONTROL_CHARACTER_STYLE));
+                    // Move the "cursor" to just after the character we're replacing
+                    last_index = idx + char.len();
+                }
+            }
+            // Append anything remaining after the last replacment
+            let remaining_line = &line[last_index..];
+            if !remaining_line.is_empty() {
+                spans.push(Span::raw(remaining_line));
+            }
```

Comment:
note: I came up with a few different ways to implement this. Ex: returning a `Vec`, implementing `Iterator`, `fold`, and experiments with `while` loops and reassigning `line` with a new slice each loop.

So far, this has been the cleanest implementation IMO, but I can share the others if interested.

It'd be great if `replacement_characters` didn't need to be called twice, but I think the match and the return type are simple enough that likely doesn't matter.